app/utility/base_planning_svc.py

```diff
@@ -44,7 +44,9 @@ class BasePlanningService(BaseService):
                 relevant_facts = await self._build_relevant_facts(variables, operation, agent_facts)
                 valid_facts = await RuleSet(rules=operation.rules).apply_rules(facts=relevant_facts[0])
                 for combo in list(itertools.product(*valid_facts)):
-                    if ability_requirements and not await self._do_enforcements(ability_requirements[link.ability.unique], operation, link, combo):
+                    if ability_requirements and not await \
+                            self._exclude_existing(combo, operation) and not await \
+                            self._do_enforcements(ability_requirements[link.ability.unique], operation, link, combo):
```

Comment:
Looking toward the future... do you foresee this conditional check "growing" with further checks?  I'm working if now would be the right time to move this check condition into it's own private function if we expect further checks.  thoughts?